engine/src/engine.rs

```diff
@@ -2281,6 +2289,18 @@ mod tests {
         assert_eq!(expected_error_kind, actual_error_kind);
     }
 
+    #[test]
+    fn test_create_legacy_address() {
+        let caller = Address::decode("3160f7328df59c14d85dfd09addad4ef18ae3e2c").unwrap();
+        let nonce = U256::from_dec_str("109438").unwrap();
+        let created_address = create_legacy_address(&caller, &nonce);
+
+        assert_eq!(
+            created_address.encode(),
+            "140e8a21d08cbb530929b012581a7c7e696145ef"
```

Comment:
Might be nice to document where this example came from.

Reply:
Transaction data included.